streams/src/main/java/org/apache/kafka/streams/processor/internals/StandbyTask.java

```diff
@@ -97,31 +96,27 @@ public void commit() {
     }
 
     @Override
-    public void close() {
-        //no-op
+    public void suspend() {
+        commit();
     }
 
     @Override
-    public void initTopology() {
-        //no-op
+    public void close() {
+        try {
+            commit();
+            closeStateManager(true);
+        } catch (final RuntimeException e) {
```

Comment:
I'm just afraid that capturing any RTE that we have not thought about and re-close the state managers may hide some issues or even subsequently trigger some other issues.